src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -694,6 +694,17 @@ private void SetGenParamCustomAttributeNoLock(CustAttr ca)
             m_ca ??= new List<TypeBuilder.CustAttr>();
             m_ca.Add(ca);
         }
+
+        internal void CreateGlobalModuleType()
+        {
+            if (!m_isHiddenGlobalType)
+            {
+                throw new NotSupportedException(SR.NotSupported_DynamicModule);
+            }
+
+            Debug.Assert(CreateTypeImpl() == null);
```

Comment:
Now this gonna run only for Debug build, I believe its needed for any build.

Anyway, I don't think asserting and checking `m_isHiddenGlobalType` is necessary, so lets not add this extra method, and call `CreateTypeImpl()` directly instead  https://github.com/dotnet/runtime/blob/a712828583b54a283d0e5343457d4b16d2234b31/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs#L855

Also the ref assembly need to be updated (remove `?`), plus add a unit test if there is no one exist that covering this scenario

Reply:
I have made your requested changes. I didn't see an obvious way to test the change to `_globalTypeBuilder.CreateTypeImpl();` in `ModuleBuilder.cs`.

Reply:
Right, `_globalTypeBuilder.CreateTypeImpl()` is not accessible directly, might add `Assert.Null(member.DeclaringType);`  into tests for global members like https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderCreateGlobalFunctions.cs or https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderDefineInitializedData.cs, but it is not necessary

Reply:
Added Assert.Null to both suite of unit tests after each call of `module.CreateGlobalFunctions();`